e2e_samples/parking_sensors/infrastructure/main.bicep

```diff
@@ -7,6 +7,8 @@ param keyvault_owner_object_id string
 @secure()
 param sql_server_password string
 param enable_monitoring bool
+param enable_keyvault_soft_delete bool = true
```

Comment:
If I remember correctly one of the parameters could be either true or null (not false) - if the value comes as null by default the parameter here will assume "true" by default because it got a null right? - but than the null won't be assumed in this case - maybe I'm missing something?

Reply:
good point. 
Short answer: I deal with this problem through the env variable and the `keyvault.bicep` file and set the purge protection option to `null` when a `false` is sent. As we decided to default to `true`, if you don't set the env variable, I am assuming `true`. If you set it to `false` I will deal with this case, and same for `true` ofc.

Long answer: Because of how the purge protection option is [implemented in Key Vault bicep](https://learn.microsoft.com/en-us/azure/templates/microsoft.keyvault/vaults?pivots=deployment-language-bicep), it won't accept `false`. I have tested and tried to send a value of `false` but it never accepted it and whenever I was sending `false` it was still enabling purge protection (as if I sent it a `true`). Until I have found that ([according to this issue](https://github.com/Azure/bicep/issues/5223)) anything you give to it that is not `null` will default to enabling purge protection. This is because by default (i.e. `null`, if you don't give it a value) it won't enable purge protection. It's quite confusing IMO. So in summary, if you want to disable purge protection, the property in bicep needs to be set to false and I deal with this in the bicep file of the key vault.

Reply:
Thanks for the comprehensive reply @elenaterenzi - indeed I though I understood it during the call, but I think now it is more clear after reading your comment :)